noodles-editor/src/noodles/fields.ts

```diff
@@ -522,7 +522,15 @@ export class BooleanField extends Field<z.ZodBoolean> {
   }
 }
 
-export class DateField extends Field<z.ZodUnion<[z.ZodDate, z.ZodISODateTime]>> {
+export class DateField extends Field<
```

Comment:
Is this now a catch-all?

Reply:
As written, I meant to rename it to `DateTime` since the field outputs a `PlainDateTime`.

It's easy to make a catch-all for the inputs. It outputs only one type and can have a pure transform.

Seems like we could either have a field for each (TimeField, DateField, DateTimeField, and ZonedDateTimeField), or maybe we could have a field that outputs the full type, a ZonedDateTime, and uses a transform to convert it to a subset type. We'd need to make sure the up and down conversation pass through with tests

Reply:
The four value types could serialize differently too. A combined field type would need to coerce the value into a zoned date time string in the project file.

I'm wondering how the filed component would reconfigure itself for each picker? The Point2D or Color field's transform are the same input for either transform, but that's not the case for the 4 types of date/times

It might be cleaner to just have 4 field types, and implement the fields to accept any input.

Is there an important reason to have one field instead, or are the types fundamentally different and so they should each have their own field?